gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/KafkaConsumerRecord.java

```diff
@@ -38,4 +38,16 @@
    * does not provide size (like Kafka 09 clients)
    */
   public long getValueSizeInBytes();
+
+  /**
+   * @return the timestamp of the underlying ConsumerRecord.
+   */
+  public default long getTimestamp() { return 0; }
```

Comment:
It might be a little bit `c/c++`-ish but shall we merge these two into a single method like `getLogAppendTime` and when the version of Kafka doesn't support LogAppendTime, returning minus value or `0`?

Reply:
The reason I kept them separate was that these methods may have usage beyond tracking record level SLA, in which case it might be useful to know what type of timestamp the ConsumerRecord is carrying.